src/dtclient/processmoduleconfig.go

```diff
@@ -21,33 +21,64 @@ type ProcessModuleProperty struct {
 	Value   string `json:"value"`
 }
 
-func (pmc *ProcessModuleConfig) AddHostGroup(hostGroup string) *ProcessModuleConfig {
+func (pmc *ProcessModuleConfig) Add(newProperty ProcessModuleProperty) *ProcessModuleConfig {
 	if pmc == nil {
-		pmc = &ProcessModuleConfig{}
+		return &ProcessModuleConfig{}
```

Comment:
if we got something empty (when the endpoint doesn't work) we still want to add the stuff the we got from other places.
this was more important when the endpoint was new, I don't know how important it is currently.
I would leave it in just in case.